### PR TITLE
Removed unnecessary protos for 0.3

### PR DESCRIPTION
**Pull request by woop**

### CODE REVIEW

File: protos/feast/core/CoreService.proto [hunk 1]
```diff
@@ -22,139 +22,49 @@ option go_package = "github.com/gojek/feast/protos/generated/go/feast/core";
...
-  // and the upload must complete in one PUT request i.e. resumable upload is not supported.
-  rpc GetUploadUrl(CoreServiceTypes.GetUploadUrlRequest) returns (CoreServiceTypes.GetUploadUrlResponse);
+    // Retrieve feature set details for a specific feature set
+    rpc GetFeatureSet (CoreServiceTypes.GetFeatureSetRequest) returns (CoreServiceTypes.GetFeatureSetResponse);
+
+    // Idempotent creation of feature set. Will not create a new feature set if schema does not change
+    rpc ApplyFeatureSet (CoreServiceTypes.ApplyFeatureSetRequest) returns (CoreServiceTypes.ApplyFeatureSetResponse);
 }
 
 message CoreServiceTypes {
```
Comment (davidheryanto) [inline protos/feast/core/CoreService.proto]: Do you think we can remove the nesting here? I feel the wrapper `CoreServiceTypes` is a bit unnecessary.

File: protos/feast/core/FeatureSet.proto [hunk 2]
```diff
@@ -16,31 +16,39 @@
...
   // Features contained within this featureSet.
-  repeated FeatureSpec features = 3;
+  repeated Feature features = 3;
+
+  // Source on which feature rows can be found
+  Source source = 4;
 }
 
-message FeatureSpec {
+message Feature {
```
Comment (davidheryanto) [inline protos/feast/core/FeatureSet.proto]: Can this cause ambiguity when we're referring to `Feature`?
Is it referering to `feast.core.Feature` or `feast.types.Feature`?
The later actually refers to a feature with name and value.

Comment (woop) [reply protos/feast/core/FeatureSet.proto]: I agree that it is a bit ambiguous, but I want to get away from using the "spec" suffix. Do you have a suggestion for a better name?

File: protos/feast/core/CoreService.proto (hunk 1 above)
Comment (woop) [reply protos/feast/core/CoreService.proto]: I am fine with either. @zhilingc thoughts?

Comment (zhilingc) [reply protos/feast/core/CoreService.proto]: Sure

File: protos/feast/core/CoreService.proto [hunk 3]
```diff
@@ -22,139 +22,49 @@ option go_package = "github.com/gojek/feast/protos/generated/go/feast/core";
...
-import "google/protobuf/timestamp.proto";
+import "feast/core/FeatureSet.proto";
 
 service CoreService {
-  // Get entities specified in request.
-  // This process returns a list of entity specs.
-  rpc GetEntities(CoreServiceTypes.GetEntitiesRequest) returns (CoreServiceTypes.GetEntitiesResponse);
-  // Get all entities
-  // This process returns a list of entity specs.
-  rpc ListEntities(google.protobuf.Empty) returns (CoreServiceTypes.ListEntitiesResponse);
```
Comment (zhilingc) [inline protos/feast/core/CoreService.proto]: Are we not having a ListFeatureSets method?

File: protos/feast/core/CoreService.proto [hunk 4]
```diff
@@ -22,139 +22,49 @@ option go_package = "github.com/gojek/feast/protos/generated/go/feast/core";
...
+    message GetFeatureSetRequest {
+        message FeatureSet {
+            string name = 1;
+
+            // (Optional) Only return details for a specific version
+            int32 version = 2;
+        }
+    }
+
+    message GetFeatureSetResponse {
```
Comment (zhilingc) [inline protos/feast/core/CoreService.proto]: method is getFeatureSet but you get a repeated value? Seems unintuitive for the user

File: protos/feast/core/FeatureSet.proto (hunk 2 above)
Comment (zhilingc) [reply protos/feast/core/FeatureSet.proto]: I think in this case `spec` is the correct term to use

File: protos/feast/core/Store.proto [hunk 5]
```diff
@@ -36,13 +35,17 @@ message ServingStore {
...
   string databaseURI = 3;
 
   // Feature sets to subscribe to. Regex strings are allowed, in the form of
-  // featureSetId:version. E.g. my-feature-set:2^
+  // featureSetId:version.
+  // Valid options for version:
+  //     latest: only subscribe to latest version of feature set
+  //     [version number]: pin to a specific version
+  //     [version number]^: subscribe to all versions larger than or equal to [version number]
   repeated string subscriptions = 4;
```
Comment (zhilingc) [inline protos/feast/core/Store.proto]: How are subscriptions for specific stores updated? How are stores added?

File: protos/feast/core/CoreService.proto (hunk 4 above)
Comment (woop) [reply protos/feast/core/CoreService.proto]: That is right. The user gets a list of feature set versions for a single feature set name. Each feature set version is an item in that list.

File: protos/feast/core/CoreService.proto (hunk 3 above)
Comment (woop) [reply protos/feast/core/CoreService.proto]: My latest proto has this. Will push.

File: protos/feast/core/Store.proto (hunk 5 above)
Comment (woop) [reply protos/feast/core/Store.proto]: This is yet to be decided. For now we are just capturing the type information

File: protos/feast/core/CoreService.proto (hunk 4 above)
Comment (zhilingc) [reply protos/feast/core/CoreService.proto]: I think that the method name should hint towards the return type

**Comment by zhilingc** [comment]
lgtm

**Pull request merged**